### PR TITLE
Fix possible NPE in Wrapper.equals

### PR DESCRIPTION
**Pull request by vbauer**

### CODE REVIEW

File: collection-matchers/src/main/java/ru/yandex/qatools/matchers/collection/Wrapper.java [hunk 1]
```diff
@@ -37,18 +37,21 @@ public int hashCode() {
     @Override
     @SuppressWarnings("unchecked")
     public boolean equals(Object obj) {
-        Wrapper<T> comparedWrapper = (Wrapper<T>) obj;
-        T toCompare = comparedWrapper.getWrapped();
+        if (obj instanceof Wrapper) {
```
Comment (lanwen) [inline collection-matchers/src/main/java/ru/yandex/qatools/matchers/collection/Wrapper.java]: think it should be guard method from guava or commons-lang3, as it error and should throw ex.

Also please add a test

Comment (vbauer) [inline collection-matchers/src/main/java/ru/yandex/qatools/matchers/collection/Wrapper.java]: > think it should be guard method from guava or commons-lang3, as it error and should throw ex.

I propose to keep it and to follow the official javadoc: http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html#equals(java.lang.Object)

> Also please add a test

Sorry, I forgot to add them - PR was updated.

Comment (lanwen) [inline collection-matchers/src/main/java/ru/yandex/qatools/matchers/collection/Wrapper.java]: ok

File: collection-matchers/src/test/java/ru/yandex/qatools/matchers/collection/WrapperTest.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+import java.util.Objects;
+
+/**
+ * @author Vladislav Bauer
+ */
+
+public class WrapperTest {
+
+    @Test
+    public void testEquals() {
```
Comment (lanwen) [inline collection-matchers/src/test/java/ru/yandex/qatools/matchers/collection/WrapperTest.java]: please use separate method for each case

File: collection-matchers/src/test/java/ru/yandex/qatools/matchers/collection/WrapperTest.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+        Wrapper<Object> objectWrapper1 = createTestWrapper();
+        Wrapper<Object> objectWrapper2 = createTestWrapper().wrap(objectWrapper1);
+        Wrapper<Object> objectWrapper3 = createTestWrapper().wrap(objectWrapper1);
+
+        //noinspection EqualsBetweenInconvertibleTypes
+        Assert.assertFalse(objectWrapper1.equals(new Object()));
```
Comment (lanwen) [inline collection-matchers/src/test/java/ru/yandex/qatools/matchers/collection/WrapperTest.java]: please add message to assertions, as of it like a smell 'assertion roulette' - we don't know which fail until look at code.

also please use `assertThat + equalTo` or `assertThat + not` form for assertions as more verbosive and refactoring friendly

Comment (vbauer) [inline collection-matchers/src/test/java/ru/yandex/qatools/matchers/collection/WrapperTest.java]: NP, I'll update it asap.

**Comment by vbauer** [comment]
PR was updated.

**Comment by lanwen** [comment]
Thanks!

**Comment by vbauer** [comment]
YW. :)

**Pull request merged**